packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -178,22 +285,71 @@ export const generateValueSpecificationForParameter = (
   );
 };
 
-const fillDerivedPropertyArguments = (
+export const fillDerivedPropertyArguments = (
   derivedPropertyExpressionState: QueryBuilderDerivedPropertyExpressionState,
-  queryBuilderState: QueryBuilderState,
+  graph: PureModel,
 ): void => {
   const propertyArguments: ValueSpecification[] =
     derivedPropertyExpressionState.parameterValues;
+  let temporalTarget: MILESTONING_STEREOTYPE | undefined;
+  if (
+    derivedPropertyExpressionState.propertyExpression.func.genericType.value
+      .rawType instanceof Class &&
+    derivedPropertyExpressionState.propertyExpression.func.owner
+      ._generatedMilestonedProperties.length !== 0
+  ) {
+    temporalTarget = getMilestoneTemporalStereotype(
+      derivedPropertyExpressionState.propertyExpression.func.genericType.value
+        .rawType,
+      derivedPropertyExpressionState.queryBuilderState.graphManagerState.graph,
+    );
+  }
   derivedPropertyExpressionState.parameters.forEach((parameter, idx) => {
     if (idx < derivedPropertyExpressionState.parameterValues.length) {
+      if (temporalTarget) {
+        const milestoningParameter = fillMilestonedDerivedPropertyArguments(
+          derivedPropertyExpressionState,
+          temporalTarget,
+          idx,
+        );
+        // Changes the `valueSpecification` to `INTERNAL__PropagatedValue` only if the parameter value
+        // matches with the corresponding `businessDate` or `processingDate`
```

Comment:
Let's also add a `NOTE: ` here to say that this will rewrite provided query as in if people explicitly specify the values, we will remove them.